crates/red_knot_python_semantic/src/import_resolution.rs

```diff
@@ -0,0 +1,104 @@
+//! Utilities for resolving import statements that are used from the `types` submodule and the `symbol` submodule.
+
+use ruff_db::files::File;
+use ruff_python_ast as ast;
+
+use crate::{
+    db::Db,
+    module_name::{ModuleName, ModuleNameResolutionError},
+    module_resolver::resolve_module,
+    semantic_index::{definition::StarImportDefinitionKind, symbol::SymbolTable},
+    symbol::SymbolAndQualifiers,
+    types::Type,
+};
+
+pub(crate) fn resolve_star_import_definition<'db>(
```

Comment:
we could possibly add some caching to this function, since it's nontrivial and it's called from both `symbol.rs` and `infer.rs`?

There's a 1% slowdown on this PR according to codspeed, which isn't huge but also isn't nothing

Reply:
I don't think this method on its own performs that much computation that it's worth caching because most its inner methods are cached (mainly `resolve_module`) and I assume it isn't called that frequently (only for star imports?)

I suspect that the perf regression is mainly due to us inferring more types. You could run the CLI in `-vvv` and see if the ingredient counts change.